### ISSUE TITLE
Dealing with delayed events when using the Cassandra plugin

### ISSUE THREAD

**Issue by chbatey** [akka/akka-projection#391]
When using the Cassandra as the source of events then events can be delayed due to:
* Clock sync
* Overload of the database
* Writes to tag_views failing and only being recovered when the persistence actor restarts

The plugin detects these with backtracking: https://github.com/akka/akka-persistence-cassandra/blob/master/core/src/main/resources/reference.conf#L345 and delivers them.

However, if there are too many missing (default 5k) it fails fast.

Projections will then restart from an offset greater than those missed events as the offset has progressed by events that weren't delayed.
These events now won't be delivered to the projection.

Instead, I think we should either:
1. Fail without restarting so the user can investigate what's going on with their database
2. Restart from a much earlier offset

For 2 we'd need to expose that information from APC via the exception so the offset is known.  
2 will result in duplicate events being delivered so the projection would need to be idempotent a.k.a effectively once via at least once with idempotent processing

**Comment by chbatey** [comment]
For context this was discovered by @mckeeh3's demo app as it was overloading the database so that writes were taking 10-60s and the eventual consistency delay was 3s.

**Comment by chbatey** [comment]
An option that makes this far less likely is to implement https://github.com/akka/akka-persistence-cassandra/issues/259 in APC. But if a 1000s of writes are taking 10s it won't fix it in all cases

**Comment by patriknw** [comment]
2 means that a CassandraSourceProvider can't be used with a JdbcProjction.exactlyOnce.

1 is difficult to implement because even if we detect this fatal situation and don't restart the projection there is a chance that the projection is started somewhere else (the original node was shutdown) and then it will still start from the saved offset.

We could try to update the offset table with some marker that this is fatally failed and requires manual intervention before starting, but there is a chance that the node crashes before that marker has been written.

Also, I'm not sure what the manual action would be if it strictly relies on exactly-once and can't handle duplicates. Rebuild the projection from scratch?
